### PR TITLE
Add Helm charts for Elasticsearch operator and cluster

### PR DESCRIPTION
**Pull request by komljen**
This is still a work in progress, but charts can be installed and everything is running fine.

The main problem is that if rbac is enabled, Helm Tiller needs to be adjusted because it doesn't have permissions to create roles.

### CODE REVIEW

**Comment by stevesloka** [comment]
Awesome! I know @snsumner has one too.

**Comment by komljen** [comment]
I wasn't aware of that one and I think that helm charts should be part of the application repository or added to official helm charts. Also, I added two separate charts for operator and ES cluster instead of combining those two because it will be much easier to maintain them and they can have separate releases.

File: charts/elasticsearch-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        readinessProbe:
+          httpGet:
+            path: /ready
+            port: 8000
+          initialDelaySeconds: 10
+          timeoutSeconds: 5
+        resources:
+{{ toYaml .Values.resources | indent 10 }}
+      imagePullSecrets:
+        - name: regsecret
```
Comment (gianrubio) [inline charts/elasticsearch-operator/templates/deployment.yaml]: This should be a variable

**Comment by gianrubio** [comment]
Can you also provide a minimal README.md?

File: charts/elasticsearch-operator/templates/deployment.yaml (hunk 1 above)
Comment (komljen) [reply charts/elasticsearch-operator/templates/deployment.yaml]: Yeah, good catch.

**Comment by komljen** [comment]
> Is it necesssary to have 2 charts for this project? Can we merge and have just one?

Yes, it is. The problem is that we could have one operator for multiple ES clusters. They also could have different releases (Helm release name). And even if we put everything in one helm chart, you can not install both with one command because CRD will not be available until operator is installed completely. Another one, if everything is in the same chart, and if you delete it, the operator will be purged, but ES cluster will remain running.

I first created one chart and saw that it is not the best idea during testing.

> Can you also provide a minimal README.md?

Sure, that is work in progress. I will add more changes today.

File: charts/elasticsearch-operator/templates/deployment.yaml (hunk 1 above)
Comment (pieterlange) [reply charts/elasticsearch-operator/templates/deployment.yaml]: It's probably even better to to add imagepullsecret references to the associated serviceaccount.
Thoughts?

**Comment by pieterlange** [comment]
I don't think it makes a lot of sense to have a chart for the `ElasticsearchCluster` resourcetype. Just exclude it? (but make a note in the operator chart that you still need to deploy the cluster)

I feel like it's another layer of indirection / templating that's not helping :)

**Comment by gianrubio** [comment]
> Yes, it is. The problem is that we could have one operator for multiple ES clusters. They also could have different releases (Helm release name). And even if we put everything in one helm chart, you can not install both with one command because CRD will not be available until operator is installed completely. Another one, if everything is in the same chart, and if you delete it, the operator will be purged, but ES cluster will remain running.

Please see how [prometheus-operator](https://github.com/coreos/prometheus-operator/blob/master/helm/prometheus-operator/templates/get-crd-job.yaml) handle the crd using a job.

**Comment by komljen** [comment]
> I feel like it's another layer of indirection / templating that's not helping :)

Yeah, I feel the same, because the cluster is just one file, but helm provides some goodies, like versioning, parameters, NOTES.txt - which is handy. I think we should probably use it for some time to see if it is good to have it or not :)

> Please see how prometheus-operator handle the crd using a job.

I saw that, but it doesn't solve the problem I described or I didn't get what you meant :) Even the prometheus operator is using a separate chart to run the cluster https://github.com/coreos/prometheus-operator/tree/master/helm/prometheus

Etcd operator, for example, is using all in one chart, but you need to run helm install and then helm upgrade to have the cluster running https://github.com/komljen/charts/blob/master/stable/etcd-operator/templates/cluster.yaml#L1

**Comment by stevesloka** [comment]
I'm on the same page with @pieterlange, I don't see a huge need for a chart for any of it, but I'm cool with merging in if that's what in the workflow for others. 

For me you need two `kubectl create` and you've got the cluster running.

**Pull request merged**